src/app/public/modules/flyout/flyout-media-query.service.ts

```diff
@@ -0,0 +1,72 @@
+import {
+  Injectable
+} from '@angular/core';
+
+import {
+  BehaviorSubject
+} from 'rxjs/BehaviorSubject';
+
+import {
+  Subscription
+} from 'rxjs/Subscription';
+
+import {
+  SkyMediaBreakpoints
+} from '@skyux/core';
+
+import {
+  SkyMediaQueryListener
+} from '@skyux/core/modules/media-query/media-query-listener';
+
+import {
+  SkyFlyoutAdapterService
+} from './flyout-adapter.service';
+
+@Injectable()
+export class SkyFlyoutMediaQueryService {
+
+  public get current(): SkyMediaBreakpoints {
+    return this._current;
+  }
+
+  private currentSubject = new BehaviorSubject<SkyMediaBreakpoints>(this.current);
+
+  private _current = SkyMediaBreakpoints.lg;
+
+  constructor(
+    private adapterService: SkyFlyoutAdapterService
+  ) {
+    this.currentSubject.next(this._current);
+  }
+
+  public subscribe(listener: SkyMediaQueryListener): Subscription {
+    return this.currentSubject.subscribe({
+      next: (breakpoints: SkyMediaBreakpoints) => {
+        listener(breakpoints);
+      }
+    });
+  }
+
+  public setBreakPoint(width: number) {
+
+    if (width <= 767) {
+      this._current = SkyMediaBreakpoints.xs;
+      this.adapterService.setFlexClass(SkyMediaBreakpoints.xs);
+    } else if (width >= 768 && width <= 991) {
+      this._current = SkyMediaBreakpoints.sm;
+      this.adapterService.setFlexClass(SkyMediaBreakpoints.sm);
+    } else if (width >= 992 && width <= 1199) {
+      this._current = SkyMediaBreakpoints.md;
+      this.adapterService.setFlexClass(SkyMediaBreakpoints.md);
+    } else {
+      this._current = SkyMediaBreakpoints.lg;
+      this.adapterService.setFlexClass(SkyMediaBreakpoints.lg);
+    }
+
+    this.currentSubject.next(this._current);
+  }
+
+  public destroy(): void {
```

Comment:
Maybe I'm missing it but does anything actually call this?

Reply:
No.... that is there because it exists on the normal media query service and so it is here for consistency with that.

Reply:
Ah, I see that now. Is this meant to be called by the consumer?

Reply:
No. I think it is meant to be called when the service is destroyed. However, services don't have the `ngOnDestroy` lifecycle hook like components do - so this is what that was replaced with.

The only place I could find where the original one was used is in unit tests.

Reply:
I've seen this pattern in other services. Looks like we intend for the consuming component to leverage it. I'm ok with it 👍

Reply:
@Blackbaud-AlexKingman @Blackbaud-TrevorBurch Services _do_ support the `ngOnDestroy`, though. https://stackoverflow.com/a/45903985/6178885

Reply:
Ah. You are correct. I was incorrectly thinking of the other lifecycle hooks such as `ngAfterViewInit`. I'm still not sure that we want to stray from what the normal media query service does in case a child component calls the `destory` method. Thoughts @Blackbaud-SteveBrush?

Reply:
We discussed this offline and decided to leave this as it is.